src/averageAges.js

```diff
@@ -36,8 +40,18 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  let women = people.filter(person => person.sex === 'f');
+
+  women = withChildren === false
+    ? women
+    : women
+      .filter(person => people
+        .some(child => child.mother === person.name));
+
+  const ages = women.map(person => person.died - person.born);
```

Comment:
you are using the same callback in 2 places
create some function to pass callback for both `map`